fw/http_sess.c

```diff
@@ -427,25 +428,29 @@ tfw_http_sticky_add(TfwHttpResp *resp, bool cache)
 	if (to_h2) {
 		set_cookie.hpack_idx = 55;
 		r = tfw_hpack_encode(resp, &set_cookie, !cache, !cache);
-	}
-	else if (cache) {
+		if (unlikely(r))
+			goto err;
+	} else if (cache) {
 		TfwHttpTransIter *mit = &resp->mit;
 		struct sk_buff **skb_head = &resp->msg.skb_head;
-		TfwStr crlf = { .data = S_CRLF, .len = SLEN(S_CRLF) };
 
 		r = tfw_http_msg_expand_data(&mit->iter, skb_head,
 					     &set_cookie, NULL);
-		if (!r)
-			r = tfw_http_msg_expand_data(&mit->iter, skb_head,
-						     &crlf, NULL);
-	}
-	else {
-		r = tfw_http_msg_hdr_add((TfwHttpMsg *)resp, &set_cookie);
+		if (unlikely(r))
+			goto err;
+		r = tfw_http_msg_expand_data(&mit->iter, skb_head, &crlf, NULL);
```

Comment:
we should also check 'r' here

Reply:
OMG. I lost, I wanted just `return r;` insted of `return 0;`.